### DISCUSSION TITLE
Any ways to solve Cloudflare R2 upload delay?

### DISCUSSION THREAD

**Discussion by Lastorder-DC** [pixelfed/pixelfed#4928] [Q&A]
```
PF_ENABLE_CLOUD=true
FILESYSTEM_CLOUD=s3
AWS_ACCESS_KEY_ID="R2-ID"
AWS_SECRET_ACCESS_KEY="R2-SECRET"
AWS_DEFAULT_REGION="auto"
AWS_BUCKET="BUCKETNAME"
AWS_URL="https://r2-connected-domain"
AWS_ENDPOINT="https://<cfid>.r2.cloudflarestorage.com/"
AWS_USE_PATH_STYLE_ENDPOINT=false
```

I tried to
1. recreate all cache(route/view/config)
2. restart nginx / pixelfed systemd
3. try to disable AWS_URL(like above) -> resulting more broken url ever(`https://maratang-arts.<cfid>.r2.cloudflarestorage.com/public/m/_v2/663158293587103745/7c649620b-69b6c5/RoLoWVDvOHYH/Vq9SIDpZ7vmOUzZANKF67Ee5WCb7fSrb0RA8PJ8n.png`)

but nothing works and image is still broken. funny thing is If I go to https://domain/i/admin/media, all images is there, but I cannot see any of them on user frontend. (url is like `https://pixelfed-domain/storage/m/_v2/663158293587103745/7c649620b-69b6c5/6efGyImQEzfZ/HJk6x7aE9HvvLjMMoMKq7EvnqGjm8tEXPA3vDVbD.png?v=1707867623`)

**Comment by Lastorder-DC** [comment]
Turns out It just takes long to upload file.(after 10 second, image appears fine)
But I still need to how to solve upload delay

**Comment by setuix** [comment]
A year later... did you figure out how to make uploads faster?

**Comment by melakhvision** [comment]
2 years later, some people face same issue with Cloudflare R2 but no solution.
